opensrp-chw/src/nacp/java/org/smartregister/chw/interactor/PncHomeVisitInteractorFlv.java

```diff
@@ -136,6 +138,48 @@ private void refreshActionList() {
         }
     }
 
+    private void evaluateVisitLocation() throws Exception {
+        HomeVisitActionHelper visitLocationHelper = new HomeVisitActionHelper() {
+            private String gpsLocation;
+
+            @Override
+            public void onPayloadReceived(String jsonPayload) {
+                try {
+                    JSONObject jsonObject = new JSONObject(jsonPayload);
+                    JSONArray fields = org.smartregister.chw.anc.util.JsonFormUtils.fields(jsonObject);
+                    gpsLocation = org.smartregister.chw.anc.util.JsonFormUtils.getFieldValue(fields, "gps");
+                } catch (JSONException e) {
+                    Timber.e(e);
+                }
+            }
+
+            @Override
+            public String evaluateSubTitle() {
+                if (gpsLocation != null && !gpsLocation.isEmpty()) {
+                    return context.getString(R.string.pnc_hv_location_captured);
+                }else{
+                    return "";
+                }
+            }
+
+            @Override
+            public BaseAncHomeVisitAction.Status evaluateStatusOnPayload() {
+                if (StringUtils.isBlank(gpsLocation))
+                    return BaseAncHomeVisitAction.Status.PENDING;
+                return BaseAncHomeVisitAction.Status.COMPLETED;
+            }
+        };
```

Comment:
I would suggest to move this implementation to a class in `actionhelper` package so as to make this file more readable.

@gosso22 What are your thoughts on this?

Reply:
@cozej4 - I with this